CONTRIBUTING/RELEASING.md

```diff
@@ -0,0 +1,534 @@
+
+You can inspect the workflows to see what they are running and copy that, but here is a general sequence of steps to mimic the automated workflow. Feel free to deviate from this as needed.
+
+1. Create a new branch from either `next` (prereleases) or `main` (patches)
+2. Get all tags: `git fetch --tags origin`
+3. `cd scripts`
+4. (If patch release) Cherry pick:
+   1. `yarn release:pick-patches`
+   2. Manually cherry pick any necessary patches based on the previous output
+5. Bump versions: `yarn release:version --verbose --release-type <RELEASE_TYPE> --pre-id <PRE_ID>`
+6. To see a list of changes (for your own to-do list), run `yarn release:generate-pr-description --current-version <CURRENT_VERSION> --next-version <NEXT_VERSION_FROM_PREVIOUS_STEP> --verbose`
+7. Write changelogs: `yarn release:write-changelog <NEXT_VERSION_FROM_PREVIOUS_STEP> --verbose`
+8. `git add .`.
+9. Commit changes: `git commit -m "Bump version from <CURRENT_VERSION> to <NEXT_VERSION_FROM_PREVIOUS_STEP> MANUALLY"`
+10. Merge changes to the release branch:
+    1. `git checkout <"latest-release" | "next-release">`
+    2. `git merge <PREVIOUS_BRANCH>`
+    3. `git push origin`
+11. (If automatic publishing is still working, it should kick in now and the rest of the steps can be skipped)
+12. `cd ..`
+13. Install dependencies: `yarn task --task=install --start-from=install`
+14. Publish to the registry: `YARN_NPM_AUTH_TOKEN=<NPM_TOKEN> yarn release:publish --tag <"next" OR "latest"> --verbose`
+15. (If patch release) `yarn release:label-patches`
+16. Manually create a GitHub Release with a tag that is the new version and the target being `latest-release` or `next-release`.
+17. Merge to core branch:
+    1. `git checkout <"next"|"main">`
+    2. `git merge <"next-release"|"latest-release">`
+    3. `git push origin`
+18. (If patch release) Sync `CHANGELOG.md` to `next` with:
+    1. `git checkout next`
+    2. `git pull`
+    3. `git checkout origin/main ./CHANGELOG.md`
+    4. `git add ./CHANGELOG.md`
+    5. `git commit -m "Update CHANGELOG.md for v<NEXT_VERSION>"`
+    6. `git push origin`
+
+## Canary Releases
+
+Not implemented yet. Still work in progress, stay tuned.
+
+## Versioning Scenarios
+
+There are seven types of releases that use the same principles, but are done somewhat differently.
+
+### Prereleases - `7.1.0-alpha.12` -> `7.1.0-alpha.13`
+
+**Cadence: Multiple times a week**
+
+This is the default strategy for prereleases, there's nothing special needed to trigger this scenario.
+
+### Prerelease promotions - `7.1.0-alpha.13` -> `7.1.0-beta.0`
+
+**Cadence: Once every 1-2 months**
+
+To promote a prerelease to a new prerelease ID, during the [Re-trigger the Workflow](#4-re-trigger-the-workflow) step, choose:
+
+- Release type: Prerelease
+- Prerelease ID: The ID to promote to. For example, for alpha to beta, write "beta".
+
+### Minor/major releases - `7.1.0-rc.2` -> `7.1.0` or `8.0.0-rc.3` -> `8.0.0`
```

Comment:
Do we always have an `rc` before each minor version release?

Reply:
I don't know, but we'll have a prerelease id for sure, and so `rc` is a fine example I think, even if it might be `beta` sometimes.